### DISCUSSION TITLE
creates documentation and search a corpus of HTML pages

### DISCUSSION THREAD

**Discussion by Roh-codeur** [squidfunk/mkdocs-material#6508] [Q&A]
hi there, this is an excellent project. I have a corpus of HTML pages. I would like to:
1. Create documentation from those. the pages are already linked to each other using HTML links.
2. Search within those HTML pages

ta!

**Comment by alexvoss** [comment]
I can only speak to (1). It is possible to use HTML in Markdown and certainly also possible to pull extra HTML into pages once they are rendered via an [MkDocs hook](https://www.mkdocs.org/user-guide/configuration/#hooks) and, say, [Beautifulsoup](https://www.crummy.com/software/BeautifulSoup/bs4/). The latter is perhaps not ideal because it involves parsing the output of Python Markdown back into a data structure in order to modify it. Won't make for the fastest build.

What approach is workable will depend more on whether you want to be able to change the HTML you are importing and how many pages we are talking about. You'd also need to take care to make them appear at the locations that are expected by the links. There is a template `redirect.html` that you can perhaps use where that is not feasible.

As to the search (2), be aware that there is a major re-write underway at the moment that may well give you what you need, see here: https://github.com/squidfunk/mkdocs-material/discussions/6490

**Comment by squidfunk** [comment]
We're actively working in improving search in #6307, and indexing HTML content alongside Markdown is on our todo list.

**Comment by Roh-codeur** [comment]
thanks for the response guys! I already have a list of set of HTML pages from my earlier website. they are all linked etc. I want to be able to upload them, run them as a website. searching these would be a key part. 

so, I understand, there is work underway to support this functionality. I will hang tight

ta!

**Reply by squidfunk** [reply]
Yes, we're working on it. Note that this sounds more like that you only want the search capabilities without Material for MkDocs, because you already have HTML pages. The new search will eventually also be Open Sourced, but this might still take some time.

**Reply by Roh-codeur** [reply]
correct! I am mostly looking for the search capability. I really like how the search works. eventually, I might even expand the website to include additional pages. happy to wait, good things come to those who do :)

ta!
